### PR TITLE
FUSETOOLS2-860/861  - Add circle-ci configuration

### PR DESCRIPTION
**Pull request by lhein**

### CODE REVIEW

**Comment by lhein** [comment]
I know we got a broken build atm but that seems to be the case for both Travis and CircleCI.

**Comment by apupier** [comment]
> I know we got a broken build atm but that seems to be the case for both Travis and CircleCI.

then need to fix the master before merging anything else

**Comment by lhein** [comment]
Yeah, I am not going to merge anything.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      - run:
+          name: test UI
+          command: npm run ui-test
+      - store_artifacts:
+          path: ./wsdl2rest.log
+      - save_cache:
+          key: dependency-cache-{{ checksum "package-lock.json" }}
+          paths:
+            - ~/.m2
+            - ~/.npm
```
Comment (apupier) [inline .circleci/config.yml:47]: given that the cache is also for .m2,, i think we need to use a key using a checksum of both package.json and pom.xml

**Review by apupier — CHANGES_REQUESTED** [review]
the maven project is not built https://github.com/camel-tooling/vscode-wsdl2rest/blob/09cc3567db484c1338e25c581ff0e16f5e5b26c1/.travis.yml#L10

**Comment by lhein** [comment]
> the maven project is not built
> 
> https://github.com/camel-tooling/vscode-wsdl2rest/blob/09cc3567db484c1338e25c581ff0e16f5e5b26c1/.travis.yml#L10

It is. Its part of the prepublish script step which is executed when you invoke `vsce package`. I agree that this is a bit weird but my goal was to leave everything as is and not break travis.

File: .circleci/config.yml (hunk 1 above)
Comment (lhein) [reply .circleci/config.yml:47]: good point. I will adapt that logic a bit.

**Comment by lhein** [comment]
build green on circle ci

**Pull request merged**